readium/navigator/src/main/java/org/readium/r2/navigator/pdf/PdfNavigatorFactory.kt

```diff
@@ -38,13 +38,15 @@ public class PdfNavigatorFactory<S : Configurable.Settings, P : Configurable.Pre
     public fun createFragmentFactory(
         initialLocator: Locator? = null,
         initialPreferences: P? = null,
-        listener: PdfNavigatorFragment.Listener? = null
+        listener: PdfNavigatorFragment.Listener? = null,
+        documentFragmentListener: L? = null
```

Comment:
I removed the `PdfEngineProvider`'s `appListener` but added a `documentFragmentListener` to handle configuration of the PDF fragments and intercepting `Resource.Exception`.

Reply:
`onResourceLoadFailed` is common between both ` PdfDocumentFragment.Listener` implementations, though it is not mentioned in the interface. I feel like it could be forwarded to `PdfNavigatorFragment.Listener`. What was wrong with having the implementation-specific listener passed to the provider constructor?